mne/inverse_sparse/mxne_optim.py

```diff
@@ -262,10 +262,74 @@ def _mixed_norm_solver_cd(M, G, alpha, maxit=10000, tol=1e-8,
 
 
 @verbose
+def _mixed_norm_solver_bcd(M, G, alpha, lipschitz_constant, maxit=200,
+                           tol=1e-8, verbose=None, init=None, n_orient=1):
+    """Solves L21 inverse problem with block coordinate descent"""
+    # First make G fortran for faster access to blocks of columns
+    G = np.asfortranarray(G)
+
+    n_sensors, n_times = M.shape
+    n_sensors, n_sources = G.shape
+    n_positions = n_sources // n_orient
+
+    if init is None:
+        X = np.zeros((n_sources, n_times))
+        R = M.copy()
+    else:
+        X = init
+        R = M - np.dot(G, X)
+
+    E = []  # track cost function
+
+    active_set = np.zeros(n_sources, dtype=np.bool)  # start with full AS
+
+    alpha_lc = alpha / lipschitz_constant
+
+    for i in range(maxit):
+        for j in range(n_positions):
+            ids = j * n_orient
+            ide = ids + n_orient
+
+            G_j = G[:, ids:ide]
+            X_j = X[ids:ide]
+
+            X_j_new = np.dot(G_j.T, R) / lipschitz_constant[j]
+
+            was_non_zero = np.any(X_j)
+            if was_non_zero:
+                R += np.dot(G_j, X_j)
+                X_j_new += X_j
+
+            block_norm = linalg.norm(X_j_new, 'fro')
+            if block_norm <= alpha_lc[j]:
+                X_j.fill(0.)
+            else:
+                shrink = np.maximum(1.0 - alpha_lc[j] / block_norm, 0.0)
+                X_j_new *= shrink
+                R -= np.dot(G_j, X_j_new)
+                X_j[:] = X_j_new
+
+        active_set = np.any(X, axis=1)
```

Comment:
I think that this is not the best way to estimate the active set. I think it would be better to set use the following:
active_set_j = active_set[ids:ide] and then setting active_set_j[:]=False if X_j = 0 and active_set_j[:]=True otherwise.

With the old way using np.any, we could obtain X.shape[0] which is not a multiple of n_orient